src/NuGet.Core/NuGet.Protocol/Utility/HttpRequestExceptionUtility.cs

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Diagnostics;
+using System.Globalization;
+using System.Net;
+using System.Net.Http;
+using NuGet.Common;
+
+namespace NuGet.Protocol.Core.Types
+{
+    internal static class HttpRequestExceptionUtility
+    {
+        internal static HttpStatusCode? GetHttpStatusCode(HttpRequestException ex)
+        {
+            HttpStatusCode? statusCode = null;
+
+#if NETCOREAPP5_0
+            statusCode = ex.StatusCode;
+#else
+            // All places which might raise an HttpRequestException need to put StatusCode in exception object.
+            if (ex.Data.Contains("StatusCode"))
```

Comment:
Promote "StatusCode" to a private static readonly field.

After deduplicating the code in `HttpSource`, you can reference the new field in that new helper method too.

Reply:
fixed in next iteration